enterprise/api/v1/views.py

```diff
@@ -188,3 +195,80 @@ class EnterpriseCustomerEntitlementViewSet(EnterpriseReadOnlyModelViewSet):
     )
     filter_fields = FIELDS
     ordering_fields = FIELDS
+
+
+class EnterpriseCatalogViewSet(viewsets.ViewSet):
+    """
+    API views for `enterprise customer catalogs` api endpoint.
+    """
+    serializer_class = serializers.EnterpriseCourseCatalogReadOnlySerializer
+    permission_classes = (permissions.IsAuthenticated,)
+    authentication_classes = (OAuth2Authentication, SessionAuthentication, BearerAuthentication, JwtAuthentication)
+    throttle_classes = (ServiceUserThrottle,)
+
+    def list(self, request):
+        """
+        DRF view to list all catalogs.
+
+        Arguments:
+            request (HttpRequest): Current request
+
+        Returns:
+            (Response): DRF response object containing course catalogs.
+        """
+        # fetch all course catalogs.
+        catalog_api = CourseCatalogApiClient(request.user)
+        catalogs = catalog_api.get_all_catalogs()
+        serializer = self.serializer_class(catalogs, many=True)
+        return Response(serializer.data)
+
+    def retrieve(self, request, pk=None):  # pylint: disable=invalid-name
```

Comment:
What's the use case for using a catalog PK as compared to an EnterpriseCustomer UUID to derive the relevant catalog? If this is EnterpriseCustomer-facing, then indexing by catalog ID means that if the EnterpriseCustomer switches to use a different catalog (for whatever reason), existing clients will keep pulling old data until they're reconfigured.